tensorflow/core/lib/strings/str_util_test.cc

```diff
@@ -417,4 +417,77 @@ TEST(StringReplace, EmptyStringReplaceAll) {
   EXPECT_EQ("", str_util::StringReplace("", "a", "X", /*replace_all=*/true));
 }
 
+TEST(SplitUTF8, Basic) {
+  std::vector<string> result;
+  // UTF8 \xE6\x82\xA8 \xE5\xA5\xBD \xE6\x82\xA8 \xE5\xA5\xBD
+  EXPECT_EQ(
+      Status::OK(),
+      str_util::SplitUTF8("\xE6\x82\xA8\xE5\xA5\xBD\xE4\xB8\x96\xE7\x95\x8C",
+                          "", false, &result));
+  EXPECT_EQ(result.size(), 4);
+
+  // UTF8 \xE6\x82\xA8 \xE5\xA5\xBD \xE4\xB8\x96 \xE7\x95\x8C
+  EXPECT_EQ(
+      Status::OK(),
+      str_util::SplitUTF8("\xE6\x82\xA8\xE5\xA5\xBD\xE4\xB8\x96\xE7\x95\x8C",
+                          "\xE5\xA5\xBD", false, &result));
+  EXPECT_EQ(result.size(), 2);
+
+  EXPECT_EQ(errors::InvalidArgument("Invalid UTF8 encoding at byte 1"),
+            str_util::SplitUTF8("\xE2\x28\xA1", "", false, &result));
+
+  EXPECT_EQ(errors::InvalidArgument(
+                "Invalid UTF8 encoding, incomplete last character"),
+            str_util::SplitUTF8("\xE6\x82", "", false, &result));
+
+  // split("###a#", "#") -> { "", "", "", "a", ""}
+  EXPECT_EQ(Status::OK(), str_util::SplitUTF8("###a#", "#", false, &result));
+  EXPECT_EQ(result.size(), 5);
+  EXPECT_EQ(result[0], "");
+  EXPECT_EQ(result[1], "");
+  EXPECT_EQ(result[2], "");
+  EXPECT_EQ(result[3], "a");
+  EXPECT_EQ(result[4], "");
+
+  // split("##a##b##c##", "#") -> {"", "", "a", "", "b", "", "c", "", ""}
+  EXPECT_EQ(Status::OK(),
+            str_util::SplitUTF8("##a##b##c##", "#", false, &result));
+  EXPECT_EQ(result.size(), 9);
+  EXPECT_EQ(result[0], "");
+  EXPECT_EQ(result[1], "");
+  EXPECT_EQ(result[2], "a");
+  EXPECT_EQ(result[3], "");
+  EXPECT_EQ(result[4], "b");
+  EXPECT_EQ(result[5], "");
+  EXPECT_EQ(result[6], "c");
+  EXPECT_EQ(result[7], "");
+  EXPECT_EQ(result[8], "");
+
+  // ASCII should work as well
+  EXPECT_EQ(Status::OK(), str_util::SplitUTF8("", ",", false, &result));
+  EXPECT_EQ(result.size(), 0);
```

Comment:
Add a test for an empty string split with `skip_empty = true` as well?

Reply:
Thanks. Done.